libraries/esp8266/examples/GPIO/FunctionalInterrupt/FunctionalInterrupt.ino

```diff
@@ -0,0 +1,85 @@
+#include <Arduino.h>
+#include <Schedule.h>
+#include "FunctionalInterrupts.h"
+
+#if defined(ESP32)
+#define BUTTON1 16
+#define BUTTON2 17
+#elif defined(ARDUINO_ESP8266_WEMOS_D1MINI)
+#define BUTTON1 D4
+#define BUTTON2 D3
+#else
+#define BUTTON1 2
+#define BUTTON2 0
+#endif
+
+class Button {
+  public:
+    Button(uint8_t reqPin) : PIN(reqPin) {
+      pinMode(PIN, INPUT_PULLUP);
+      // Arduino C API:
+      attachInterruptArg(PIN, [](void* self) {
+        static_cast<Button*>(self)->isr();
+      }, this, FALLING); // works on ESP32; fails on ESP8266: "ISR not in IRAM"
+      //attachInterruptArg(PIN, reinterpret_cast<void(*)(void*)>(&isr_static), this, FALLING); // works on ESP32; works on ESP8266
+      // FunctionalInterrupts API:
+      //attachInterrupt(PIN, [this]() { isr(); }, FALLING); // works on ESP32; works on ESP8266
+      //attachScheduledInterrupt(PIN, [this](InterruptInfo ii) { Serial.print("Pin "); Serial.println(ii.pin); isr(); }, FALLING); // works on ESP32; works on ESP8266
```

Comment:
This is a valid use of std::bind, std::function or lambdas, since the actual call is made in the context of the task that runs the scheduler, not in a interrupt service routine.